### PR TITLE
Feat: Add access control to LyricsFlipNFT contract minting

### LINKED ISSUE #107: CONTRACT: Add Access Control to `LyricsFlipNFT` contract
**Description:**
Add AccessControl using OpenZeppelin's AccessControl component to LyricsFlipNFT contract. Restrict the mint functionality to be called only by the `LyricsFlip` contract. Write a comprehensive test to cover all edge cases.

### PR DESCRIPTION
**Pull request by psychemist**
# 🎶  LyricsFlip Pull Request

- [x] Closes #107 
- [x] Added tests
- [x] Run tests
- [x] Run formatting
- [x] Evidence attached
- [x] Commented the code

---

### 📌 Type of Change

- [ ] Documentation (updates to README, docs, or comments)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

---

## 📝 Changes description

- Added OpenZeppelin access control to the LyricsFlipNFT contract minting function to prevent minting from any origin that is not the LyricsFlip contract
- Added a robust unit testing suite to confirm mint from a specified origin and verify hitch-free mint to recipients 

---

## 📸 Evidence 

Screenshot 2025-02-22 at 20 30 25

- `scarb test`

Screenshot 2025-02-22 at 20 33 59

- `scarb build`

---

## 🌌 Comments

- Verify LyricsFlipNFT does not have a predetermined address and works with any deployed contract instance

### CODE REVIEW

**Comment by psychemist** [comment]
@manlikeHB please check changes and merge

**Review by manlikeHB — CHANGES_REQUESTED** [review]

File: onchain/src/tests/test_lyricsflipNFT.cairo [hunk 1]
```diff
@@ -39,34 +85,210 @@ fn setup_dispatcher() -> (ContractAddress, NFTDispatcher) {
...
+    // Verify can still mint after transfer
+    start_cheat_caller_address(nft_address, lyrics_flip_address);
+    dispatcher.mint(recipient());
+    stop_cheat_caller_address(nft_address);
+
+    assert(erc721.owner_of(2) == recipient(), 'Wrong owner of new token');
+}
+
+#[test]
+fn test_mint_to_zero_address_should_fail() {
```
[the remark this answers is not in the document]
Comment (psychemist) [reply onchain/src/tests/test_lyricsflipNFT.cairo:240]: will fix this

**Review by manlikeHB — APPROVED** [review]

**Comment by manlikeHB** [comment]
Thank you @psychemist! LGTM!

**Comment by psychemist** [comment]
@manlikeHB i've effected more changes and tests implementing SRC5 interface that check the minter is a deployed LyricsFlip contract. please review
